handlers/meshery_application_handler.go

```diff
@@ -40,7 +50,14 @@ func (h *Handler) ApplicationFileHandler(
 	user *models.User,
 	provider models.Provider,
 ) {
-	// Application files are just pattern files
+	// Applications can be- pattern, k8s manifests,etc. By default if no "format" is present in headers, it will be treated as a pattern file.
+	format := r.Header.Get("format")
```

Comment:
Nit: we might want to ask the client to send this information in the body.

Reply:
just like what we are doing [here](https://github.com/meshery/meshery/blob/master/handlers/meshery_pattern_handler.go#L199)

Reply:
Given today's discussion, I'll withhold approval until the approach is consistent.

Reply:
Given the second approach, we have everything already on backend. UI would need to make different API calls and this will fix that. So I am closing the pull request.